.github/workflows/tests-cygwin.yml

```diff
@@ -56,7 +56,6 @@ jobs:
           TMP: "/tmp"
           TEMP: "/tmp"
         run: |
-          cd "${GITHUB_WORKSPACE}"
```

Comment:
The code that fails is this https://github.com/mesonbuild/meson-python/blob/ef67e0e03c3224a8703809ce6eafb8ac5ce352d5/tests/conftest.py#L33-L51 which is called for example here https://github.com/mesonbuild/meson-python/blob/ef67e0e03c3224a8703809ce6eafb8ac5ce352d5/tests/conftest.py#L96
It should not depend on CWD of the tests runner. Why it fails although is not very clear.

Reply:
I was going to say "GitPython hasn't done a release since they merged the changes to get it working on Cygwin", but apparently that's only true on GitHub; PyPI had two releases a month ago.

`actions/checkout` is done with Windows git, not Cygwin, so there would be (different) errors operating on the `meson-python` repository, but the errors are apparently from a brand-new repository, so that's not it either.

My next guess would be `fork()` failure since CI runs into that a decent bit, but this project doesn't have many dependencies, nor does it have many C extension modules, and also I think I've found the actual `stdout` and `stderr` in the logs, which show a different message.

There's no reasons coming to mind for why `rm -rf .git; git init; git config gpgsign false` would fail, especially with that error

Reply:
Indeed. By the way @DWesl , do you know if there is a n easy way to cache the cygwin installation done by the `cygwin/cygwin-install-action`? Debugging this on the CI having to wait for Cygwin to install at each run is a bit tedious. On a branch I setup caching for the Python dependencies, cutting down the `pip install` time considerably, but installing Cygwin still takes anything between one minute and a half and three minutes.

Reply:
The fix for the `git` issue is this: https://github.com/mesonbuild/meson-python/blob/68e1d391e5a81d588da35ebaabe908473dbcbfee/.github/workflows/tests.yml#L91-L98

Reply:
So it's the second issue, but git for some reason decided to have two error messages for that.  Lovely